### PR TITLE
Derek/android raw

### PR DESCRIPTION
**Pull request by betaBison**
- differentiates the Google Smartphone Decimeter Challenge into `google_decimeter.py` and GNSSLogger parsers into `android.py`
- updates unit_test directories according to the above change
- adds parsers for data obtained from the GNSSLogger Android app including`Raw`, `Fix` and all sensors
- adds recommended filters for raw GNSS data from android phones
- adds new NavData classes for the 2023 version of the Google Smartphone Decimeter Challenge
- updates python package dependencies
- minor bug fixes
- minor doc updates

### CODE REVIEW

File: tests/parsers/test_android.py [hunk 1]
```diff
@@ -115,246 +102,107 @@ def fixture_raw_path(root_path):
...
     # raises exception if not a file path
     with pytest.raises(FileNotFoundError):
-        android.AndroidRawImu("not_a_file.txt")
+        sensor_type("not_a_file.txt")
     with pytest.raises(FileNotFoundError):
-        android.AndroidRawImu(pathlib.Path("not_a_file.txt"))
+        sensor_type(pathlib.Path("not_a_file.txt"))
 
     # raises exception if input not string or path-like
     with pytest.raises(TypeError):
```
Comment (danineamati) [inline tests/parsers/test_android.py:204]: We probably also want to test that the returned navdata is not empty, not nan, and that the data values match the expected result.

File: gnss_lib_py/parsers/android.py [hunk 2]
```diff
@@ -8,401 +18,532 @@
...
-        # convert bearing degrees to heading in radians
-        self["heading_rx_gt_rad"] = np.deg2rad(self["heading_rx_gt_rad"])
-        self["heading_rx_gt_rad"] = wrap_0_to_2pi(self["heading_rx_gt_rad"])
+        # add heading in radians
+        self["heading_rx_rad"] = np.deg2rad(self["heading_rx_deg"])
 
     @staticmethod
     def _row_map():
-        """Map row names from loaded data to gnss_lib_py standard
+        """Map of column names from loaded to gnss_lib_py standard
```
Comment (danineamati) [inline gnss_lib_py/parsers/android.py:492]: There may be a notation mismatch here.

File: gnss_lib_py/parsers/android.py [hunk 3]
```diff
@@ -422,278 +562,169 @@ def preprocess(self, input_path):
...
-        """Read Android raw file and produce location fix dataframe objects
+        # add gps milliseconds
+        self["gps_millis"] = unix_to_gps_millis(self["unix_millis"])
 
-        Parameters
-        ----------
-        input_path : string or path-like
-            File location of data file to read.
+    def _row_map(self):
+        """Map of column names from loaded to gnss_lib_py standard
```
Comment (danineamati) [inline gnss_lib_py/parsers/android.py:604]: Similarly, here the docstring has `column` but elsewhere variables are labelled as `row`.

File: notebooks/timing_comparisons.ipynb
Comment (danineamati) [inline notebooks/timing_comparisons.ipynb:1]: Will the path location update with the pull request?

File: tests/parsers/test_google_decimeter.py [hunk 4]
```diff
@@ -0,0 +1,823 @@
...
+
+def test_navdata_type(derived):
+    """Test that all subclasses inherit from NavData
+
+    Parameters
+    ----------
+    derived : pytest.fixture
+        Instance of AndroidDerived2021 for testing
+    """
+    isinstance(derived, NavData)
```
Comment (danineamati) [inline tests/parsers/test_google_decimeter.py:341]: Is an assert statement needed here? If not, perhaps an assert statement would improve clarity?

File: tests/parsers/test_google_decimeter.py [hunk 5]
```diff
@@ -0,0 +1,823 @@
...
+    """Test AndroidGroundTruth initialization
+
+    Parameters
+    ----------
+    gtruth : AndroidGroundTruth2021
+        Instance of AndroidGroundTruth2021 for testing
+
+    """
+
+    isinstance(gtruth, NavData)
```
Comment (danineamati) [inline tests/parsers/test_google_decimeter.py:445]: Similarly, an assert statement may be useful here for clarity.

File: tests/parsers/test_android.py [hunk 6]
```diff
@@ -115,246 +102,107 @@ def fixture_raw_path(root_path):
...
-
-    Parameters
-    ----------
-    android_raw_path : pytest.fixture
-        Path to Android Raw measurements text log file
-    """
-    test_imu = android.AndroidRawImu(android_raw_path)
-    isinstance(test_imu, NavData)
+    test_navdata = sensor_type(input_path = raw_path)
+    isinstance(test_navdata, NavData)
```
Comment (danineamati) [inline tests/parsers/test_android.py:192]: Perhaps an assert statement here would be helpful.

File: notebooks/timing_comparisons.ipynb
Comment (betaBison) [reply notebooks/timing_comparisons.ipynb:1]: Excellent question. No, the path doesn't change. Our strategy in the past has been to point to an existing file until the pull request review and then change to the `main` branch. If we changed to the `main` branch right now the docs wouldn't build since that file doesn't exist in the correct location in the main branch.

File: tests/parsers/test_android.py (hunk 1 above)
Comment (betaBison) [reply tests/parsers/test_android.py:204]: additional test added and bug caught!

File: gnss_lib_py/parsers/android.py (hunk 2 above)
Comment (betaBison) [reply gnss_lib_py/parsers/android.py:492]: updated 10 places in the repo

File: gnss_lib_py/parsers/android.py (hunk 3 above)
Comment (betaBison) [reply gnss_lib_py/parsers/android.py:604]: updated 10 places in the repo

File: tests/parsers/test_google_decimeter.py (hunk 4 above)
Comment (betaBison) [reply tests/parsers/test_google_decimeter.py]: an assert statement _is_ needed there or else nothing happens. This test was effectively pointless. Updated 13 places throughout the repo

File: tests/parsers/test_google_decimeter.py (hunk 5 above)
Comment (betaBison) [reply tests/parsers/test_google_decimeter.py]: Updated 13 places throughout the repo

File: tests/parsers/test_android.py (hunk 6 above)
Comment (betaBison) [reply tests/parsers/test_android.py]: Updated 13 places throughout the repo

**Review by danineamati — APPROVED** [review]
All comments were resolved, and the test passed on separate machines. I will double-check the new file locations after the merge.
